entrypoint.sh

```diff
@@ -263,9 +262,21 @@ configure_nvidia_installation_dirs() {
   popd
 }
 
+installer_default_download_url() {
+  # projects/000000000000/zones/us-west1-a -> us
+  local -r instance_location="$(curl -sfS "http://metadata.google.internal/computeMetadata/v1/instance/zone" -H "Metadata-Flavor: Google" | cut -d '/' -f4 | cut -d '-' -f1)"
+  declare -A location_mapping
+  location_mapping=( ["us"]="us" ["asia"]="asia" ["europe"]="eu" )
+  # Use us as default download location.
+  local -r download_location="${location_mapping[${instance_location}]:-us}"
+  echo ${NVIDIA_DRIVER_DOWNLOAD_URL_DEFAULT} | sed "s/\${NVIDIA_DRIVER_DOWNLOAD_LOCATION}/${download_location=}/"
+}
+
 download_nvidia_installer() {
   info "Downloading Nvidia installer ... "
   pushd "${NVIDIA_INSTALL_DIR_CONTAINER}"
+  NVIDIA_DRIVER_DOWNLOAD_URL="${NVIDIA_DRIVER_DOWNLOAD_URL:-$(installer_default_download_url)}"
```

Comment:
This can be a local var now.
I suggest do this:

```
# define a function to get the actual download url:
get_installer_download_url() {
  echo "${NVIDIA_DRIVER_DOWNLOAD_URL:-$(installer_default_download_url)}"
}

# define a function to get the runfile name
get_installer_runfile() {
  local url="$(get_installer_download_url)"
  echo "$(basename "$url")"
}
```

Use these function replacing `NVIDIA_DRIVER_DOWNLOAD_URL` and `NVIDIA_INSTALLER_RUNFILE`. And remove the global variables.